cli/cmd/replicaset.go

```diff
@@ -634,14 +630,14 @@ func internalReplicasetDowngradeModule(cmdCtx *cmdcontext.CmdCtx, args []string)
 	}, replicasetcmd.DowngradeOpts{
 		ChosenReplicasetAliases: chosenReplicasetAliases,
 		Timeout:                 lsnTimeout,
-		DowngradeVersion:        downgradeVersion,
+		DowngradeVersion:        args[1],
```

Comment:
You were trying to avoid overuse of `args`, but why not use here the named variable `downgradeVersion` and set it at the `PreRun` step?

Reply:
It seems there is some misunderstanding.
My main concern is to avoid using args as an _argument in a signature_ of a function that is not command handler, because when I see a function that expects `args` it is unclear the exact sequence of `args` that are expected. And even if it's documented, it would look like unnecessary constraint. Keeping `args` argument only in command handlers forces us to put the knowledge about every arg in `args` only in corresponding handler and don't spread out it to helper functions.
Considering the above your suggestion is the next step to improve readability. We may 'resolve' `args` to named variables in the very beginning of the handler and then refer them instead of `args`. I would prefer to declare it locally rather than using global variables (which would needed if declaring it in `PreRun`). I know that all (or at least the most of) optional arguments (flags in Cobra's terms) are represented with global variables, but it is forced by Cobra's flags parsing machinery which is not applied to positional arguments.

Reply:
I've made change to illustrate what I mean